packages/daemon/src/app/controller/job.ts

```diff
@@ -60,19 +61,21 @@ export class JobController {
         sse.finish();
       }
     } else {
-      try {
-        const plugins = await this.pipelineService.installPlugins(job, cwd, pyIndex);
-        this.pipelineService.startJob(job, cwd, plugins);
-      } catch (err) {
-        return failRes(this.ctx, {
-          message: err?.message,
-          data: job
-        });
-      }
       successRes(this.ctx, {
         message: 'create pipeline and jobs successfully',
         data: job
       }, 201);
+      process.nextTick(async () => {
+        let plugins: Partial<Record<PluginTypeI, PluginInfo>>;
+        try {
+          plugins = await this.pipelineService.installPlugins(job, cwd, pyIndex);
+        } catch (err) {
+          job.status = PipelineStatus.FAIL;
+          job.error = err.message;
+          return job.save();
+        }
+        this.pipelineService.startJob(job, cwd, plugins);
```

Comment:
Why do we move the `startJob()` outside of the try-clause?

Reply:
`startJob` do not throw any error.